code/game/machinery/kitchen/cooking_machines/_appliance.dm

```diff
@@ -389,14 +389,15 @@
 		CI.result_type = 4//Recipe type, a specific recipe will transform the ingredients into a new food
 		var/list/results = recipe.make_food(C)
 
-		var/atom/temp = new /atom //To prevent infinite loops, all results will be moved into a temporary location so they're not considered as inputs for other recipes
+		var/obj/temp = new /obj(src) //To prevent infinite loops, all results will be moved into a temporary location so they're not considered as inputs for other recipes
 
 		for (var/atom/movable/AM in results)
 			AM.loc = temp
 
 		//making multiple copies of a recipe from one container. For example, tons of fries
 		while (select_recipe(available_recipes,C) == recipe)
-			var/list/TR = recipe.make_food(C)
+			var/list/TR = list()
+			TR.Add(recipe.make_food(C))
```

Comment:
this is robustness. make_food should always return a list but in some of the overridden versions it didn't.